lib/spack/spack/environment.py

```diff
@@ -776,8 +777,8 @@ def included_config_scopes(self):
         # highest-precedence scopes are last.
         includes = config_dict(self.yaml).get('include', [])
         for i, config_path in enumerate(reversed(includes)):
-            # allow paths to contain environment variables
-            config_path = config_path.format(**os.environ)
+            # allow paths to contain spack config/environment variables, etc.
+            config_path = substitute_path_variables(config_path)
```

Comment:
```suggestion
            config_path = spack.util.path.canonicalize_path(config_path)
```
`canonicalize_path` is usually a better entry-point than `substitute_path_variables`. In this case the only difference is it would take the absolute path.

Reply:
I actually did try `canonicalize_path` first (see [the diff for the first commit](https://github.com/spack/spack/pull/16210/commits/3f85fc577db8a269b1ce18e4eaa2da306a28e1d1)), but it created issues with the `test_env_with_included_config_file` regression test (see [here](https://travis-ci.org/github/spack/spack/jobs/677788099)). This was caused by `canonicalize_path` expanding `.` to a different directory than the test case directory, which in turn caused the test to be unable to locate the referenced config file. It looks to me like the [code below my patch](https://github.com/spack/spack/blob/develop/lib/spack/spack/environment.py#L787) handles expanding `.` to the correct path, so using `substitute_path` instead seems appropriate.